contracts/multisig-prover/src/execute.rs

```diff
@@ -20,10 +20,16 @@ use crate::{
 pub fn construct_proof(
     deps: DepsMut,
     env: Env,
-    message_ids: Vec<String>,
+    message_ids: Vec<CrossChainId>,
 ) -> Result<Response, ContractError> {
     let config = CONFIG.load(deps.storage)?;
-    let batch_id = BatchID::new(&message_ids, None);
+    let batch_id = BatchID::new(
```

Comment:
the constructor of BatchId is a bit strange. Why take a slice reference just to clone it into a new vector and sort it? I suggest changing the constructor to accept ownership of `Vec<CrossChainId>`

Reply:
furthermore, I don't understand why it accepts a worker set. That case is never used

Reply:
I changed the constructor to just accept a slice reference of CrossChainId, and removed the clone. We convert to string right away anyways, so we don't need to take ownership.

The worker set case is actually used by `CommandBatchBuilder::build`